File: src/main/java/cookbook/backend/UserProfileBackend.java
package cookbook.backend;

public class UserProfileBackend {

    // 🌟 PROFILE MEMORY BANK
    public static String firstName = "Guest";
    public static String lastName = "";
    public static String email = "guest@example.com";

    /**
     * Extracts the first letter of the user's first name for the avatar.
     */
    public static String getAvatarInitial() {
        if (firstName == null || firstName.trim().isEmpty()) {
            return "";
        }
        return firstName.substring(0, 1).toUpperCase();
    }

    /**
     * Updates the profile information.
     */
    public static void updateProfile(String newFirstName, String newLastName) {
        firstName = (newFirstName != null) ? newFirstName.trim() : "";
        lastName = (newLastName != null) ? newLastName.trim() : "";
    }

    /**
     * Clears the user session data upon logout.
     */
    public static void performLogout() {
        firstName = "Guest";
        lastName = "";
        email = "guest@example.com";
    }
}

File: app/src/main/java/com/cookbook/backend/UserProfileBackend.kt
package com.cookbook.backend

object UserProfileBackend {
    var firstName: String = "Guest"
    var lastName: String = ""
    var email: String = ""

    fun getAvatarInitial(): String {
        return if (firstName.isNotBlank() && firstName != "Guest") {
            firstName.first().uppercase()
        } else "G"
    }

    fun updateProfile(first: String, last: String) {
        firstName = first
        lastName = last
    }

    fun performLogout() {
        FirebaseManager.logout()
        firstName = "Guest"
        lastName = ""
        email = ""
    }
}
